packages/signer/signer-btc/src/internal/app-binder/task/UpdatePsbtTask.ts

```
@@ -0,0 +1,297 @@
+import {
+  ByteArrayBuilder,
+  type CommandResult,
+  CommandResultFactory,
+  UnknownDeviceExchangeError,
+} from "@ledgerhq/device-management-kit";
+import { Either, EitherAsync } from "purify-ts";
+
+import { type Psbt as ApiPsbt } from "@api/model/Psbt";
+import { type BtcErrorCodes } from "@internal/app-binder/command/utils/bitcoinAppErrors";
+import { type PsbtSignature } from "@internal/app-binder/task/SignPsbtTask";
+import {
+  type Psbt as InternalPsbt,
+  PsbtGlobal,
+  PsbtIn,
+} from "@internal/psbt/model/Psbt";
+import { Value } from "@internal/psbt/model/Value";
+import { type PsbtMapper } from "@internal/psbt/service/psbt/PsbtMapper";
+import { type ValueParser } from "@internal/psbt/service/value/ValueParser";
+import { encodeVarint } from "@internal/utils/Varint";
+
+type UpdatePsbtTaskArgs = {
+  psbt: ApiPsbt;
+  signatures: PsbtSignature[];
+};
+
+export class UpdatePsbtTask {
+  constructor(
+    private readonly _args: UpdatePsbtTaskArgs,
+    private readonly _valueParser: ValueParser,
+    private readonly _psbtMapper: PsbtMapper,
+  ) {}
+
+  public async run(): Promise<CommandResult<InternalPsbt, BtcErrorCodes>> {
```

[ASK] detail but does it really need to be async? I think that task dont make device interactions nor network calls?

---

I used async flow to handle easily thrown errors from private methods with `EitherAsync` instead of verbose Either return in for loops